### PR TITLE
fix(anthropic): forward unmapped request fields to Chat Completions b…

### PR DESCRIPTION
**Pull request by szedan-rh**
## Problem

The Anthropic Messages to Chat Completions translation rebuilt the request from an allowlist and silently dropped every other field. A client sending a provider-specific parameter got a normal-looking response with that parameter never delivered. `top_k` only survived through a hand-written special case.

## Change

Every top-level request field now falls into one of four buckets, following the rule agreed on #385 and refined in review:

| Bucket | Behavior | Fields |
| --- | --- | --- |
| Mapped | Translated to the Chat Completions equivalent | existing set, plus `metadata.user_id` → SHA-256 hex `safety_identifier`, `output_config.effort` → `reasoning_effort`, `output_config.format` (`json_schema`, or the deprecated `output_format`) → strict `response_format` |
| Rejected | 400 `invalid_request_error` naming the field | `service_tier`, `container`, `inference_geo`, `mcp_servers`, and any unmapped `output_config` key such as `task_budget` |
| Dropped | Removed with a log warning | `thinking`, `context_management` |
| Forwarded | Sent untouched, backend validates | everything else, including `top_k` |

The reject list is the set of fields whose effect the translated response would otherwise fabricate (`wire.rs` hardcodes them to null). The drop list is the two fields Claude Code sends on every request that have no Chat Completions equivalent and whose absence the response reports truthfully. A translated key always wins over a colliding client key. The `top_k` special case is removed.

`safety_identifier` is sent as the SHA-256 digest because Anthropic allows 512 characters and null while Chat allows 64 non-null. `response_format` carries `strict: true` because Anthropic structured outputs guarantee schema conformance.

## Handled

- Anthropic Messages → Chat Completions request translation.
- Unit tests for each bucket, the hashing, the `output_config` mapping, and rejection of unmapped `output_config` keys.
- Two synthetic inference fixtures recorded through the importer: `messages/provider-parameter-passthrough` (forward, map, drop) and `messages/unrepresentable-parameters` (reject with no upstream call). Coverage manifest and README regenerated.
- An Anthropic SDK test against a stub backend, runnable with `uv run`, wired into CI as the `anthropic-sdk` job.
- Filter doc and the Anthropic guide describe the rule.

## Not handled

- Responses → Chat Completions has the same allowlist defect. Follow-up PR, now that #1176 has landed its reject helper.
- Response direction. The Anthropic response types are fixed structs, agreed out of scope on #385.
- `cache_control` is forwarded. Moving it is one entry in `UNREPRESENTABLE_FIELDS` if preferred.
- Bedrock, Vertex and Cohere translators belong to #114.

## Verified

`cargo test -p praxis-ai-apis`, `make lint`, `cargo xtask check-inference`, `make test-inference-fixtures`, the SDK test via `uv run`, and the pinned Claude Code acceptance test with `PRAXIS_TEST_CLAUDE_CODE_BIN` set all pass locally.

Part of #385

### CODE REVIEW

**Comment by szedan-rh** [comment]
CI caught a real conflict with the reject list, so the follow-up commit adjusts it.

The pinned Claude Code acceptance test failed with a 400 on `thinking`. I ran the real CLI against a stub `/v1/messages` to see what it actually sends, and on every request it includes:

- `thinking` (`{"type":"enabled","budget_tokens":31999,...}` or `{"type":"adaptive"}`)
- `context_management` (a single `clear_thinking` edit)
- `output_config` (`{"effort":"high"}`, plus a JSON-schema `format` on the title request)

Rejecting any of these locks Claude Code out of the translation path entirely. Forwarding them is not an option either: vLLM ignores unknown fields but the OpenAI API returns 400, so the same request would work or fail depending on the backend.

These three also do not meet the reject criterion we agreed on. The translated response carries no thinking blocks, so `thinking` is visibly absent rather than fabricated, and `context_management` only edits thinking blocks that do not exist on this path. So the rule stays as agreed, with one honest exception: fields a mainstream client sends unconditionally, with no Chat Completions equivalent, whose absence degrades quality rather than correctness, are dropped with a warning.

Final buckets:

- Reject (400): `service_tier`, `container`, `inference_geo`, `mcp_servers`
- Drop with warning: `thinking`, `context_management`, `output_config`, `output_format`
- Map: `metadata.user_id` → `safety_identifier`
- Forward: everything else

Mapping `output_config.format` to `response_format` is a real feature and I'll open a follow-up issue for it rather than fold it in here.

Verified locally with `PRAXIS_TEST_CLAUDE_CODE_BIN` set: the pinned Claude Code workflow test passes, along with the unit tests, fixture replay, and the new SDK test.

**Review by leseb — CHANGES_REQUESTED** [review]
[P1] output_config is silently ignored. [DROPPED_FIELDS](https://github.com/praxis-proxy/ai/blob/6904348b96c4db5d53cffe882eca35ffc3654450/apis/src/anthropic/messages_to_chat_completions/request.rs#L23-L34) removes structured-output and effort requirements, yet returns a normal response. This violates the agreed [#385](https://github.com/praxis-proxy/ai/issues/385) behavior; map output_config.format or reject it. The Claude Code test masks this by hardcoding schema-compliant output.

[P2] metadata.user_id crosses incompatible contracts. [map_metadata](https://github.com/praxis-proxy/ai/blob/6904348b96c4db5d53cffe882eca35ffc3654450/apis/src/anthropic/messages_to_chat_completions/request.rs#L680-L684) forwards null and identifiers up to Anthropic’s 512-character limit into Chat Completions’ non-null, 64-character safety_identifier. Valid Anthropic requests can therefore fail upstream. Omit null and hash, constrain, or explicitly reject oversized values.

[P2] The new SDK test has no effective CI coverage. Its documented command fails with ModuleNotFoundError: conftest, and neither Make nor CI invokes the file. Running it correctly through pytest passes two tests, but that path is not automated.

**Comment by szedan-rh** [comment]
@leseb Thank you for the review.
handled.

**Review by leseb — APPROVED** [review]
Approving but please followup with those:

* n > 1 is forwarded, but both response translators return only choices[0], silently discarding additional billed outputs.
* Valid output_config.task_budget: null is rejected instead of treated as unset.

File: apis/src/anthropic/messages_to_chat_completions/request.rs [hunk 1]
```diff
@@ -595,21 +659,87 @@ fn convert_stream(chat: &mut Map<String, Value>, stream: Option<Value>, stream_o
...
+        Some(key) => Err(format!(
+            "`output_config.{key}` is not supported when translating Anthropic Messages to Chat Completions"
+        )),
+        None => Ok(()),
+    }
+}
+
+/// Build a strict Chat Completions `json_schema` response format from an
+/// Anthropic `json_schema` output format.
+fn response_format(format: Value) -> Result<Value, String> {
```
[the remark this answers is not in the document]
Comment (szedan-rh) [reply apis/src/anthropic/messages_to_chat_completions/request.rs:732]: Declining this one. `{"type": "text"}` is not an Anthropic output format. The vendored spec in `docs/conformance/specs/anthropic-spec.json` defines `output_config.format` as `JsonOutputFormat | null`, and the only format schema (stable and beta) has `type` fixed to `json_schema`. The Anthropic SDK 1.6.0 likewise has a single `JSONOutputFormatParam` with `type: Literal["json_schema"]`, and the structured-outputs docs show no other type. The `text` variant is OpenAI's `ResponseFormatText`, not Anthropic's.

**Review by nerdalert — APPROVED** [review]
The earlier substantive review items are addressed:

  - output_config is mapped or rejected rather than silently ignored.
  - metadata.user_id is hashed into a valid 64-character safety_identifier.
  - The Anthropic SDK test is now wired into CI.
  - The bot’s type: text concern does not match Anthropic’s documented schema.

The approved follow-ups around n > 1 response handling and `task_budget: null` are valid edge cases, but neither appears introduced by this PR or warrants blocking it.
